packages/x-date-pickers/src/internals/hooks/useMaskedInput.tsx

```diff
@@ -59,44 +57,46 @@ export const useMaskedInput = <TInputDate, TDate>({
   );
 
   // TODO: Implement with controlled vs uncontrolled `rawValue`
-  const currentInputValue = getDisplayDate(utils, rawValue, inputFormat);
-  const [innerInputValue, setInnerInputValue] = React.useState(currentInputValue);
-  const previousInputValueRef = React.useRef(currentInputValue);
-  React.useEffect(() => {
-    previousInputValueRef.current = currentInputValue;
-  }, [currentInputValue]);
-  const notTyping = !isFocused;
-  const valueChanged = previousInputValueRef.current !== currentInputValue;
-  // Update the input value only if the value changed outside of typing
-  if (notTyping && valueChanged && (rawValue === null || utils.isValid(rawValue))) {
-    if (currentInputValue !== innerInputValue) {
-      setInnerInputValue(currentInputValue);
-    }
+  const parsedValue = rawValue === null ? null : utils.date(rawValue);
+
+  // Track the value of the input
+  const [innerInputValue, setInnerInputValue] = React.useState(parsedValue);
```

Comment:
Can you set the explicit generic on `useState` for those two states ?
It would be easier to read since it's a tricky part of the codebase